be/ut/alloc.c

```diff
@@ -192,6 +192,145 @@ M0_INTERNAL void m0_be_ut_alloc_concurrent(void)
+	 * Get credits and allocate memory. Confirm the memory allocated with
+	 * m0_be_alloc_aligned() is filled with zeroes. Fill this memory with
+	 * some non-zero data and release this memory. Again allocate memory and
+	 * confirm the same memory pointer is returned by the allocator and the
+	 * memory is zeroed out.
+	 * Next get the credits and allocate memory with no-clear. Confirm the
+	 * memory allocated using m0_be_alloc_aligned is NOT filled with zeroes.
+	 */
+	size = 4096;
+	shift = 0;
+	cred = empty_cred;
+	m0_be_allocator_credit(a, M0_BAO_ALLOC_ALIGNED, size, shift, &cred);
+	m0_be_allocator_credit(a, M0_BAO_FREE_ALIGNED, size, shift, &cred);
+	m0_be_allocator_credit(a, M0_BAO_ALLOC_ALIGNED, size, shift, &cred);
+	m0_be_allocator_credit(a, M0_BAO_FREE_ALIGNED, size, shift, &cred);
+	m0_be_allocator_credit_no_clear(a, M0_BAO_ALLOC_ALIGNED, size, shift,
+					&cred);
+	m0_be_allocator_credit_no_clear(a, M0_BAO_FREE_ALIGNED, size, shift,
+					&cred);
+	m0_be_allocator_credit_no_clear(a, M0_BAO_ALLOC_ALIGNED, size, shift,
+					&cred);
+	m0_be_allocator_credit_no_clear(a, M0_BAO_FREE_ALIGNED, size, shift,
+					&cred);
+
+	m0_be_ut_tx_init(&tx, ut_be);
+	m0_be_tx_prep(&tx, &cred);
+	rc = m0_be_tx_open_sync(&tx);
+
+	M0_BE_OP_SYNC(op, m0_be_alloc_aligned(a, &tx, &op, &p, size,
+					      shift, M0_BITS(M0_BAP_NORMAL),
+					      false));
+	M0_ASSERT(p != NULL);
+	saved_p = p;
+	for (i=0; i < size/sizeof(uint64_t); i++) {
+		M0_ASSERT(((uint64_t *)p)[i] == 0);
+		((uint64_t *)p)[i] = i;
+	}
+
+	M0_BE_OP_SYNC(op, m0_be_free_aligned(a, &tx, &op, p));
+
+	M0_BE_OP_SYNC(op, m0_be_alloc_aligned(a, &tx, &op, &p, size,
+					      shift, M0_BITS(M0_BAP_NORMAL),
+					      false));
+	/**
+	 * We know the allocator uses the LIFO scheme when dispersing memory to
+	 * its callers. Based on this we can assume to get the recently released
+	 * memory back with the allocation call above.
+	 */
+	M0_ASSERT(p != NULL && p == saved_p);
+	for (i=0; i < size/sizeof(uint64_t); i++) {
+		M0_ASSERT(((uint64_t *)p)[i] == 0);
+	}
+	M0_BE_OP_SYNC(op, m0_be_free_aligned(a, &tx, &op, p));
+
+
+	M0_BE_OP_SYNC(op, m0_be_alloc_aligned_no_clear(a, &tx, &op, &p, size,
+						       shift,
+						       M0_BITS(M0_BAP_NORMAL),
+						       false));
+	M0_ASSERT(p != NULL);
+	for (i=0; i < size/sizeof(uint64_t); i++) {
+		M0_ASSERT(((uint64_t *)p)[i] == 0);
+		((uint64_t *)p)[i] = i;
+	}
+
+	M0_BE_OP_SYNC(op, m0_be_free_aligned(a, &tx, &op, p));
+
+	M0_BE_OP_SYNC(op, m0_be_alloc_aligned_no_clear(a, &tx, &op, &p, size,
+						       shift,
+						       M0_BITS(M0_BAP_NORMAL),
+						       false));
+	M0_ASSERT(p != NULL && p == saved_p);
+	for (i=0; i < size/sizeof(uint64_t); i++) {
+		M0_ASSERT(((uint64_t *)p)[i] == i);
+	}
+	M0_BE_OP_SYNC(op, m0_be_free_aligned(a, &tx, &op, p));
+
+	m0_be_tx_close_sync(&tx);
+	m0_be_tx_fini(&tx);
+	m0_be_ut_seg_allocator_fini(ut_seg, ut_be);
+	m0_be_ut_seg_fini(ut_seg);
+	m0_be_ut_backend_fini(ut_be);
+	M0_SET0(ut_be);
+
+
+	tx = tx;
```

Comment:
Is this required?

Reply:
This is to suppress compiler warning about variable not used.